core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnInsertExec.scala

```diff
@@ -78,7 +82,134 @@ case class ColumnInsertExec(_child: SparkPlan, partitionColumns: Seq[String],
     else ("1", "0x0")
   }
 
+  def loop(code: String, numTimes: Int): String = {
+    s"""
+      for(int i=0;i< $numTimes; i++){
+          $code
+      }""".stripMargin
+  }
+
+  /**
+    * This method will be used when column count exceeds 30 to avoid
+    * the 64K size limit of JVM. Most of the code which generated big codes, has been
+    * chaunked or put in an array/row to avoid huge code blocks. This will impact the performance,
+    * but will work till the store column limit of 1012.
+    */
+  private def doProduceWideTable(ctx: CodegenContext): String = {
+    val encodingClass = classOf[ColumnEncoding].getName
+    val numInsertedRowsMetric = if (onExecutor) null
+    else metricTerm(ctx, "numInsertedRows")
+    schemaTerm = ctx.addReferenceObj("schema", relationSchema,
+      classOf[StructType].getName)
+
+    val schemaLength = relationSchema.length
+    encoderArrayTerm = ctx.freshName("encoderArray")
+    cursorArrayTerm = ctx.freshName("cursorArray")
+    encoderCursorTerms = relationSchema.map { _ =>
+      (ctx.freshName("encoder"), ctx.freshName("cursor"))
+    }
+    numInsertions = ctx.freshName("numInsertions")
+    ctx.addMutableState("long", numInsertions, s"$numInsertions = -1L;")
+    maxDeltaRowsTerm = ctx.freshName("maxDeltaRows")
+    batchSizeTerm = ctx.freshName("currentBatchSize")
+    val batchSizeDeclaration = if (true) {
+      ctx.addMutableState("int", batchSizeTerm, s"$batchSizeTerm = 0;")
+      ""
+    } else {
+      s"int $batchSizeTerm = 0;"
+    }
+    defaultBatchSizeTerm = ctx.freshName("defaultBatchSize")
+    ctx.addMutableState("int", defaultBatchSizeTerm, "")
+    val defaultRowSize = ctx.freshName("defaultRowSize")
+    val childProduce = doChildProduce(ctx)
+
+    child match {
+      case c: CallbackColumnInsert =>
+        ctx.addNewFunction(c.resetInsertions,
+          s"""
+             |public final void ${c.resetInsertions}() {
+             |  $batchSizeTerm = 0;
+             |  $numInsertions = -1;
+             |}
+          """.stripMargin)
+        batchBucketIdTerm = Some(c.bucketIdTerm)
+      case _ =>
+    }
+
+    val initEncoderCode =
+      s"""
+         |this.$encoderArrayTerm[i] = $encodingClass$$.MODULE$$.getColumnEncoder(
+         |           |  $schemaTerm.fields()[i]);
+       """.stripMargin
+
+    val initEncoderArray = loop(initEncoderCode, relationSchema.length)
+    val iniCursorCode = s"this.$cursorArrayTerm[i] = 0L;"
+    val iniCursorArray = loop(iniCursorCode, relationSchema.length)
+
+    ctx.addMutableState("org.apache.spark.sql.execution.columnar.encoding.ColumnEncoder[]",
```

Comment:
use embedded class variable rather than hardcoded class path:

```scala
  val encoderClass = classOf[ColumnEncoder].getName
  ... 
  ctx.addMutableState(s"$encoderClass[]", ...)
```